cmd/genji/insert.go

```diff
@@ -0,0 +1,140 @@
+package main
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/dgraph-io/badger/v2"
+	"github.com/genjidb/genji"
+	"github.com/genjidb/genji/document"
+	"github.com/genjidb/genji/engine"
+	"github.com/genjidb/genji/engine/badgerengine"
+	"github.com/genjidb/genji/engine/boltengine"
+)
+
+func skipSpaces(r *bufio.Reader) (byte, error) {
+	var c byte
+	var err error
+
+	for {
+		c, err = r.ReadByte()
+		if err != nil {
+			return c, err
+		}
+
+		if c != '\n' && c != '\r' && c != ' ' && c != '\t' {
+			break
+		}
+	}
+
+	return c, nil
+}
+
+func executeInsertCommand(db *genji.DB, table string, r io.Reader) error {
+	q := fmt.Sprintf("INSERT INTO %s VALUES ?", table)
+	rd := bufio.NewReader(r)
+	var c byte
+	var err error
+
+	// Ignore spaces.
+	c, err = skipSpaces(rd)
+	if err != nil {
+		return err
+	}
+	switch c {
+	case '{': // Json objects
```

Comment:
This only works for one json object, a json stream is a **multiple** json objects, you need to look until the reader is over. You have an example here: https://golang.org/pkg/encoding/json/#Decoder